ccm_web/client/src/api.ts

```diff
@@ -55,18 +65,23 @@ export const getGlobals = async (key: string | undefined): Promise<Globals> => {
   return await resp.json()
 }
 
-const delay = async (ms: number): Promise<void> => {
-  await new Promise<void>(resolve => setTimeout(() => resolve(), ms))
+// usage:
+//   const data = await delay(nnnn).then(() => {return something})
+// const delay = async (ms: number): Promise<void> => {
+//   await new Promise<void>(resolve => setTimeout(() => resolve(), ms))
+// }
+
+export const getCourseSections = async (key: string | undefined, courseId: number): Promise<CanvasCourseSection[]> => {
+  const request = getGet(key)
+  const resp = await fetch('/api/course/' + courseId.toString() + '/sections', request)
+  await handleErrors(resp)
+  return await resp.json()
 }
 
-// This is a placeholder for a real implementation (I mean, obviously :D)
-export const getCourseSections = async (key: string | undefined, courseId: string): Promise<string[]> => {
-  const sections = await delay(2000).then(() => {
-    if (Math.random() * 3 > 1) {
-      return (['AAAA', 'BBBB'])
-    } else {
-      return new Promise<string[]>((resolve, reject) => { reject(new Error('Error retrieving course section information.')) })
-    }
-  })
-  return sections
+export const addCourseSections = async (key: string | undefined, courseId: number, sectionNames: string[]): Promise<CanvasCourseSection[]> => {
+  const body = JSON.stringify({ sections: sectionNames })
+  const request = getPost(key, body)
+  const resp = await fetch('/api/course/' + courseId.toString() + '/sections', request)
+  await handleErrors(resp)
+  return await resp.json()
```

Comment:
I see this error in the browser console when getting sections are loading. Probably I feel this is due the way the promise is handled
```
react-dom.development.js:67 Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of Transition which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node
    at div
    at Transition (https://925849cc5a54.ngrok.io/main-00f62ec4d542a7626326.js:57029:30)
    at Fade (https://925849cc5a54.ngrok.io/main-00f62ec4d542a7626326.js:3573:24)
    at Backdrop (https://925849cc5a54.ngrok.io/main-00f62ec4d542a7626326.js:632:24)
    at WithStyles (https://925849cc5a54.ngrok.io/main-00f62ec4d542a7626326.js:19027:31)
    at div
    at div
    at div
    at BulkSectionCreate (https://925849cc5a54.ngrok.io/main-00f62ec4d542a7626326.js:64493:21)
```

Reply:
I think this is from a 3rd party library but I can check again

Reply:
I found what appears to be a solution for it, though I think it is simply just conditionally using a workaround in development.  I'm not sure if it makes it go away in production or not.

However, it is an issue with a 3rd party library and material-ui so it doesn't really belong in this issue.

The workaround involves using the  ThemeProvider so it might be best handled as part of issue #75  for implementing an application level theme.

I moved the SnackbarProvider up to Index.tsx from App.tsx and wrapped it in ThemeProvider.
This solution is talked about [here](https://github.com/mui-org/material-ui/issues/13394#issuecomment-815452717) on a material-ui issue.

```
import { createMuiTheme, ThemeProvider, unstable_createMuiStrictModeTheme } from '@material-ui/core'

const ltiKey = getLTIKey()

const createTheme = process.env.NODE_ENV === 'production' ? createMuiTheme : unstable_createMuiStrictModeTheme
const theme = createTheme({

})

ReactDOM.render(
  <React.StrictMode>
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={3}>
        <App ltiKey={ltiKey} />
      </SnackbarProvider>
    </ThemeProvider>
  </React.StrictMode>,
  document.getElementById('root')
)
```